sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -65,6 +66,8 @@ private[sql] class JacksonGenerator(
       (arr: SpecializedGetters, i: Int) => {
         writeObject(writeMapData(arr.getMap(i), mt, mapElementWriter))
       }
+    case _ => throw new UnsupportedOperationException(
+      s"Initial type ${dataType.catalogString} must be an array, a struct or a map")
```

Comment:
nit: `s"Initial type ${dataType.catalogString} must be an array, a struct or a map")` -> `s"Initial type ${dataType.catalogString} must be an ${ArrayType.simpleString}, a ${StructType.simpleString} or a ${MapType.simpleString}")`?

Reply:
Changed here and above